zerver/views/message_edit.py

```diff
@@ -121,6 +121,21 @@ def update_message_backend(request: HttpRequest, user_profile: UserMessage,
             content = "(deleted)"
         content = normalize_body(content)
 
+    # Check for no-op changes; if we find any, ignore them and set a flag.
+    # The purpose of the flag is to allow us to return json_success() in the
+    # case that changes were provided but no changes ended up being made, due
+    # to all of them being no-op changes.
```

Comment:
Can you explain a bit of the intent/reasoning around which cases should result in an error verse a silent success?

Reply:
I was working from what you said [here](https://chat.zulip.org/#narrow/stream/206-zulip-terminal/topic/topic.20editing.20-.20time.20limit/near/904998). I'll expand the comment there to explain.